test/matrix.cc

```diff
@@ -391,6 +393,142 @@ TEST(Matrix, partial_matrix) {
   EXPECT_EQ(found, 2) << " partial result did not find 2 results as expected";
 }
 
+// slim dowm matrix response: https://github.com/valhalla/valhalla/pull/3987
+const auto test_matrix_default = R"({
+    "sources":[
+      {"lat":52.103948,"lon":5.06813}
+    ],
+    "targets":[
+      {"lat":52.106126,"lon":5.101497},
+      {"lat":52.100469,"lon":5.087099},
+      {"lat":52.103105,"lon":5.081005},
+      {"lat":52.094273,"lon":5.075254}
+    ],
+    "costing":"auto"
+  })";
+
+TEST(Matrix, default_matrix) {
+  loki_worker_t loki_worker(config);
+
+  Api request;
+  ParseApi(test_matrix_default, Options::sources_to_targets, request);
+
+  loki_worker.matrix(request);
+
+  GraphReader reader(config.get_child("mjolnir"));
+
+  sif::mode_costing_t mode_costing;
+  mode_costing[0] =
+      CreateSimpleCost(request.options().costings().find(request.options().costing_type())->second);
+
+  TimeDistanceMatrix timedist_matrix;
+  std::vector<TimeDistance> results =
+      timedist_matrix.SourceToTarget(*request.mutable_options()->mutable_sources(),
+                                     *request.mutable_options()->mutable_targets(), reader,
+                                     mode_costing, sif::TravelMode::kDrive, 400000.0);
+
+  auto response = serializeMatrix(request, results, 1.0);
```

Comment:
these tests are too verbose, they recreate a bunch of the stuff that happens in the internal library for no purpose. dont make your own workers and your own graph reader unless you need to have them for the test. at the end of the day all you care about in your test is what the json looks like. so just make one call to `tyr::actor_t::matrix` and parse the json as you do below. something like this should be enough:
```suggestion
  tyr::actor_t actor(config, true);
  auto response = actor.matrix(request);
```

Reply:
It's defined like this:
```c++
actor_t::matrix(const std::string& request_str, const std::function<void()>* interrupt, Api* api)
```
We can't use request i.e, Api here

Reply:
sorry i used the wrong variable name. ive updated the suggestion to use the string defined above